### PR TITLE
Some more code cleanup (V3.0)

### PR DESCRIPTION
**Pull request by Nothing4You**

### CODE REVIEW

**Comment by Nothing4You** [comment]
Seems like mono doesn't like syntactic sugar :(
I'll look into what can be used later, don't merge for now.

File: src/IniFileParser/Model/Formatting/IniDataFormatter.cs [hunk 1]
```diff
@@ -55,8 +55,8 @@ public virtual string IniDataToString(IniData iniData)
         /// </remarks>
         public IniFormattingConfiguration Format
         {
-            get { return _configuration; }
-            set { _configuration = value.Clone(); }
+            get => _configuration;
+            set => _configuration = value.Clone();
```
Comment (Svetomech) [inline src/IniFileParser/Model/Formatting/IniDataFormatter.cs]: I don't think this is valid.

File: src/IniFileParser/Model/IniData.cs [hunk 2]
```diff
@@ -91,8 +91,8 @@ public IniData(IniData ori)
         /// </summary>
         public SectionDataCollection Sections
         {
-            get { return _sections; }
-            set { _sections = value; }
+            get => _sections;
+            set => _sections = value;
```
Comment (Svetomech) [inline src/IniFileParser/Model/IniData.cs]: Same as above.

File: src/IniFileParser/Model/KeyData.cs [hunk 3]
```diff
@@ -51,17 +51,17 @@ public KeyData(KeyData ori)
         /// </summary>
         public List<string> Comments
         {
-            get { return _comments; }
-            set { _comments = new List<string> (value) ; }
+            get => _comments;
+            set => _comments = new List<string> (value);
```
Comment (Svetomech) [inline src/IniFileParser/Model/KeyData.cs]: Same.

File: src/IniFileParser/Model/KeyData.cs [hunk 4]
```diff
@@ -51,17 +51,17 @@ public KeyData(KeyData ori)
...
 
         /// <summary>
         ///     Gets or sets the value associated to this key.
         /// </summary>
         public string Value
         {
-            get { return _value; }
-            set { _value = value; }
+            get => _value;
+            set => _value = value;
```
Comment (Svetomech) [inline src/IniFileParser/Model/KeyData.cs]: Not valid.

File: src/IniFileParser/Model/SectionData.cs [hunk 5]
```diff
@@ -126,45 +126,25 @@ public string SectionName
 		[Obsolete("Do not use this property, use property Comments instead")]
         public List<string> LeadingComments
         {
-            get
-            {
-                return _leadingComments;
-            }
+            get => _leadingComments;
```
Comment (Svetomech) [inline src/IniFileParser/Model/SectionData.cs]: Not valid.

File: src/IniFileParser/Model/SectionData.cs [hunk 6]
```diff
@@ -126,45 +126,25 @@ public string SectionName
...
 
-            internal set
-            {
-                _leadingComments = new List<string>(value);
-            }
-        }
+		    internal set => _leadingComments = new List<string>(value);
```
Comment (Svetomech) [inline src/IniFileParser/Model/SectionData.cs]: Not valid.

File: src/IniFileParser/Model/SectionData.cs [hunk 7]
```diff
@@ -126,45 +126,25 @@ public string SectionName
...
 
-		[Obsolete("Do not use this property, use property Comments instead")]
+        [Obsolete("Do not use this property, use property Comments instead")]
         public List<string> TrailingComments
         {
-            get
-            {
-                return _trailingComments;
-            }
+            get => _trailingComments;
```
Comment (Svetomech) [inline src/IniFileParser/Model/SectionData.cs]: Not valid.

File: src/IniFileParser/Model/SectionData.cs [hunk 8]
```diff
@@ -126,45 +126,25 @@ public string SectionName
...
 
-            internal set
-            {
-                _trailingComments = new List<string>(value);
-            }
+            internal set => _trailingComments = new List<string>(value);
```
Comment (Svetomech) [inline src/IniFileParser/Model/SectionData.cs]: Not valid.

File: src/IniFileParser/Model/SectionData.cs [hunk 9]
```diff
@@ -174,15 +154,9 @@ internal set
         /// </value>
         public KeyDataCollection Keys
         {
-            get
-            {
-                return _keyDataCollection;
-            }
+            get => _keyDataCollection;
```
Comment (Svetomech) [inline src/IniFileParser/Model/SectionData.cs]: Not valid.

File: src/IniFileParser/Model/SectionData.cs [hunk 10]
```diff
@@ -174,15 +154,9 @@ internal set
...
 
-            set
-            {
-                _keyDataCollection = value;
-            }
+            set => _keyDataCollection = value;
```
Comment (Svetomech) [inline src/IniFileParser/Model/SectionData.cs]: Not valid.

File: src/IniFileParser/Parser/ConcatenateDuplicatedKeysIniDataParser.cs [hunk 11]
```diff
@@ -11,14 +8,8 @@ public class ConcatenateDuplicatedKeysIniDataParser : IniDataParser
...
-            get
-            {
-                return (ConcatenateDuplicatedKeysIniParserConfiguration)base.Configuration;
-            }
-            set
-            {
-                base.Configuration = value;
-            }
+            get => (ConcatenateDuplicatedKeysIniParserConfiguration)base.Configuration;
+            set => base.Configuration = value;
```
Comment (Svetomech) [inline src/IniFileParser/Parser/ConcatenateDuplicatedKeysIniDataParser.cs]: Not valid.

File: src/IniFileParser/Parser/IniDataParser.cs [hunk 12]
```diff
@@ -350,7 +350,7 @@ protected virtual void ProcessKeyValuePair(string line, IniData currentIniData)
             // get key and value data
             string key = ExtractKey(line);
 
-			if (string.IsNullOrEmpty(key) && Configuration.SkipInvalidLines) return;
+            if (string.IsNullOrEmpty(key) && Configuration.SkipInvalidLines) return;
```
Comment (Svetomech) [inline src/IniFileParser/Parser/IniDataParser.cs:353]: I'd suggest using `String.IsNullOrWhiteSpace` since it also includes emptiness check.

**Comment by Nothing4You** [comment]
@Svetomech it's valid syntactic sugar, works fine with roslyn / C# 6.0 but appearently mono doesn't like that syntactic sugar. all this code compiles perfectly fine in VS 2017.
See https://msdn.microsoft.com/en-us/magazine/dn802602.aspx (ctrl+f `Expression Bodied Functions and Properties`, second match)

**Comment by Svetomech** [comment]
@Nothing4You I'm perfectly aware of all C# 6.0 syntactic sugar. Still, my point stands - you're not using it correctly. `=>` can not be used immediately after `get` or `set`.

File: src/IniFileParser/Parser/IniDataParser.cs [hunk 13]
```diff
@@ -68,10 +68,10 @@ public IniDataParser(IniParserConfiguration parserConfiguration)
         /// exception that was raised.
         /// </remarks>
 
-        public ReadOnlyCollection<Exception> Errors {get {return _errorExceptions.AsReadOnly();} }
-		#endregion
+        public ReadOnlyCollection<Exception> Errors => _errorExceptions.AsReadOnly();
```
Comment (Svetomech) [inline src/IniFileParser/Parser/IniDataParser.cs:71]: E.g. this is correct.

**Comment by Svetomech** [comment]
@Nothing4You My guess the places I pointed are causing the build to fail.

**Comment by Nothing4You** [comment]
Since VS2017 compiles the entire project without any errors I'm pretty sure that it is valid syntax:
`========== Rebuild All: 3 succeeded, 0 failed, 0 skipped ==========`

**Comment by Svetomech** [comment]
@Nothing4You Of course it does. VS 2017 comes bundled with C# 7.0 support which [has extended expression bodied members](https://blogs.msdn.microsoft.com/dotnet/2017/03/09/new-features-in-c-7-0/).

**Comment by Nothing4You** [comment]
@Svetomech thanks for that, i'll reduce it to what is supported in C# 6.0 tomorrow, which hopefully fixes the mono issues.

**Comment by Svetomech** [comment]
@Nothing4You No problem. I'm almost certain it will.

**Comment by rickyah** [comment]
Thanks @Svetomech for the review. I jumped of the .net wagon on c# 4.5   and although I knew about the new syntax sugar and some additions of 6.0 this is still a little bit alien to me :D

**Comment by Svetomech** [comment]
My pleasure, @rickyah. I had a great time using your library in one of my projects and will use it again if need be. So it only seems fair to me that I contribute, even in a small way (:

File: src/IniFileParser/Parser/IniDataParser.cs (hunk 12 above)
Comment (Nothing4You) [reply src/IniFileParser/Parser/IniDataParser.cs:353]: That change should be done in a separate commit. this PR isn't intended to change behavior.

**Comment by Nothing4You** [comment]
So appearently Mono 3.12.1 does not fully support C# 6.0 either.
Do you plan on keeping Mono 3.12.1 as a supported platform?
The only part currently where Mono 3.12.1 complains is [this](https://github.com/Nothing4You/ini-parser/blob/f7ca114f1e16d02ae890e78b02c50dc7698cf094/src/IniFileParser/Exceptions/ParsingException.cs#L14), where I added some comments about Mono 3.12.1 compatibility.

File: src/IniFileParser/Parser/IniDataParser.cs (hunk 12 above)
Comment (rickyah) [reply src/IniFileParser/Parser/IniDataParser.cs:353]: No need to do it. `ExtractKey()` return a trimmed string anyway

Comment (Svetomech) [reply src/IniFileParser/Parser/IniDataParser.cs:353]: Then I also remembered it was only introduced in .Net 4.0, so... yeah, forget the suggestion altogether.

**Comment by Svetomech** [comment]
That's really strange, there should be no problems with get-only properties in mono.

**Comment by Nothing4You** [comment]
See https://travis-ci.org/rickyah/ini-parser/builds/222870097

**Comment by Nothing4You** [comment]
It doesn't seem to like that it's initialized in the constructor instead of the line defining the property

**Comment by Svetomech** [comment]
> Auto-implemented property `IniParser.Exceptions.ParsingException.LibVersion' must have set accessor or initializer

Oh, I see. But this is the way it was before C# 6.0 for sure. It appears its support is partial in the 3.12.1 version

**Comment by Nothing4You** [comment]
Yeah, Mono 3 seems to have partial C# 6.0 support. As of Mono 4 C# 6.0 should be fully supported: http://www.mono-project.com/docs/about-mono/releases/4.0.0/

**Pull request closed without merging**